src/functional/ffront/parsing/foir_to_itir.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from typing import List, Optional
+
+from eve import NodeTranslator
+from functional.ffront import field_operator_ir as foir
+from functional.iterator import ir as iir
+
+
+class SymExprResolver(NodeTranslator):
+    @classmethod
+    def parse(cls, nodes: List[foir.Expr], *, params: Optional[list[iir.Sym]] = None) -> foir.Expr:
```

Comment:
I would standardize the name of the classmethod meant to be used as entry point for all visitor-like classes (parsers, passes, etc.) as `apply()`, as I think it is done right in GT4Py & GTC